oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/OAuth2AuthorizationCodeRequestAuthenticationContext.java

```diff
@@ -78,6 +101,20 @@ public static Builder with(OAuth2AuthorizationCodeRequestAuthenticationToken aut
 	 */
 	public static final class Builder extends AbstractBuilder<OAuth2AuthorizationCodeRequestAuthenticationContext, Builder> {
 
+		/**
+		 * Associates an attribute.
+		 *
+		 * @param key the key for the attribute
+		 * @param value the value of the attribute
+		 * @return the {@link Builder} for further configuration
+		 */
+		@Override
+		public Builder put(Object key, Object value) {
```

Comment:
> This can be removed since it's inherited by `AbstractBuilder`

I overrode this method as opposed to using the one in AbstractBuilder because the one in the AbstractBuilder does not accept null values and `authorizationConsent` can be null.

The alternative is to remove the not null assertion inside the AbstractBuilder, but it's inherited 5 times so I didn't want to cause potential issues in other builders.

Reply:
If `authorizationConsent` or `authorizationRequest` is `null` it should not be passed into the `builder` so it's not an issue.

Reply:
> If `authorizationConsent` or `authorizationRequest` is `null` it should not be passed into the `builder` so it's not an issue.

The `authorizationConsent` from `authorizationConsentService.findById()` is `@Nullable`.

So I added a null check which won't pass it into the builder if it's null. I got rid of the custom `put()` method.
Let me know if the solution is suitable.